packages/crud/src/vaadin-crud.js

```diff
@@ -921,73 +910,97 @@ class Crud extends SlotMixin(ControllerMixin(ElementMixin(ThemableMixin(PolymerE
     form.addEventListener('input', this.__changeListener);
   }
 
-  /** @private */
-  __onSaveButtonChange(save, old) {
-    this.__setupSlottedButton(save, old, this.__saveBound);
-  }
-
-  /** @private */
-  __onDeleteButtonChange(deleteButton, old) {
-    this.__setupSlottedButton(deleteButton, old, this.__deleteBound);
+  /**
+   * @param {HTMLElement | undefined} form
+   * @param {string} theme
+   * @param {string | string[] | undefined} include
+   * @param {string | RegExp} exclude
+   * @private
+   */
+  __formPropsChanged(form, theme, include, exclude) {
+    if (form) {
+      form.include = include;
+      form.exclude = exclude;
+      form.setAttribute('theme', theme);
+    }
   }
 
-  /** @private */
-  __onCancelButtonChange(cancel, old) {
-    this.__setupSlottedButton(cancel, old, this.__cancelBound);
+  /**
+   * @param {HTMLElement} saveButton
+   * @param {HTMLElement} oldSaveButton
+   * @private
+   */
+  __onSaveButtonChange(saveButton, oldSaveButton) {
+    this.__setupSlottedButton(saveButton, oldSaveButton, this.__saveBound);
   }
 
-  /** @private */
-  __setupSlottedButton(slottedButton, currentButton, clickListener) {
-    if (currentButton && currentButton.parentElement) {
-      currentButton.parentElement.removeChild(currentButton);
+  /**
+   * @param {HTMLElement | undefined} saveButton
+   * @param {string} i18nLabel
+   * @param {boolean} isDirty
+   * @private
+   */
+  __saveButtonPropsChanged(saveButton, i18nLabel, isDirty) {
+    if (saveButton) {
+      saveButton.toggleAttribute('disabled', this.__isSaveBtnDisabled(isDirty));
+      saveButton.textContent = i18nLabel;
     }
-
-    slottedButton.addEventListener('click', clickListener);
   }
 
-  /** @private */
-  __hostPropsChanged() {
-    this.__propagateHostAttributes();
+  /**
+   * @param {HTMLElement} deleteButton
+   * @param {HTMLElement} oldDeleteButton
+   * @private
+   */
+  __onDeleteButtonChange(deleteButton, oldDeleteButton) {
```

Comment:
```suggestion
  __onDeleteButtonChanged(deleteButton, oldDeleteButton) {
```

Reply:
Done in https://github.com/vaadin/web-components/pull/3676.